src/Nethermind/Nethermind.Synchronization/Blocks/BlockDownloader.cs

```diff
@@ -346,26 +346,16 @@ private ValueTask DownloadFailHandler<T>(Task<T> downloadTask, string entities)
             if (downloadTask.IsFaulted)
             {
                 _sinceLastTimeout = 0;
-                if (downloadTask.Exception?.InnerException is TimeoutException
-                    || (downloadTask.Exception?.InnerExceptions.Any(x => x is TimeoutException) ?? false)
-                    || (downloadTask.Exception?.InnerExceptions.Any(x => x.InnerException is TimeoutException) ?? false))
+                if (downloadTask.Exception?.Flatten().InnerExceptions.Any(x => x is TimeoutException) ?? false)
                 {
                     if (_logger.IsTrace) _logger.Error($"Failed to retrieve {entities} when synchronizing (Timeout)", downloadTask.Exception);
                     _syncBatchSize.Shrink();
                 }
-                else
-                {
-                    Exception exception = downloadTask.Exception;
-                    AggregateException aggregateException = exception as AggregateException;
-                    if (aggregateException != null)
-                    {
-                        exception = aggregateException.Flatten().InnerExceptions[0];
-                    }
 
-                    if (_logger.IsInfo) _logger.Error($"Failed to retrieve {entities} when synchronizing.", exception);
+                if (downloadTask.Exception != null)
+                {
+                    _ = downloadTask.Result; // trying to throw with stack trace
```

Comment:
Huh?

You might want this: https://docs.microsoft.com/en-us/dotnet/api/system.runtime.exceptionservices.exceptiondispatchinfo.throw?view=netcore-3.1#System_Runtime_ExceptionServices_ExceptionDispatchInfo_Throw_System_Exception_

Reply:
nice to know although after analysing I think it would achieve the same result

Reply:
this is me saying I give up on these tests in this commit (although some parts of them are still relevant so better to keep them this way)